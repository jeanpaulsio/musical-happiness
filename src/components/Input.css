.input-container {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

input {
  width: 50%;
  border: 0;
  border-bottom: 3px solid #dddddd;
  padding-bottom: 5px;
  font-size: 20px;
  transition: border-bottom 0.1s ease-in;
}

input:focus, input:hover {
  outline: none;
  border-bottom: 3px solid #36393d;
}
