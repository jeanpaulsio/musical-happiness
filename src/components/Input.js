import React from "react";
import { func, string } from "prop-types";
import "./Input.css";

Input.propTypes = {
  value: string.isRequired,
  placeholder: string,
  onChange: func.isRequired
};

function Input(props) {
  return (
    <div className="input-container">
      <input
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
      />
    </div>
  );
}

export default Input;
