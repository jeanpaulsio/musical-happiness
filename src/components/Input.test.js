import React from "react";
import renderer from "react-test-renderer";
import Input from "./Input";

test("<Input /> renders correctly", () => {
  const value = "value";
  const onChange = () => {};
  const tree = renderer
    .create(<Input value={value} onChange={onChange} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

