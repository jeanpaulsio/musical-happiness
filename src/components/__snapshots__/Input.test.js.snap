// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<div
  className="input-container"
>
  <input
    onChange={[Function]}
    placeholder={undefined}
    value="value"
  />
</div>
`;
